anndata/compat/__init__.py

```diff
@@ -62,12 +62,37 @@ class Literal(metaclass=LiteralMeta):
             pass
 
 
+def _read_hdf5_attribute(attrs: h5py.AttributeManager, name: str):
+    """
+    Read an HDF5 attribute and perform all necessary conversions.
+
+    At the moment, this only implements conversions for string attributes, other types
+    are passed through. String conversion is needed compatibility with other languages.
+    For example Julia's HDF5.jl writes string attributes as fixed-size strings, which
+    are read as bytes by h5py.
+    """
+    attr = attrs[name]
+    attr_id = attrs.get_id(name)
+    dtype = h5py.check_string_dtype(attr_id.dtype)
+    if dtype is None:
+        return attr
+    else:
+        if dtype.length is None:  # variable-length string, no problem
+            return attr
+        elif len(attr_id.shape) == 0:  # Python bytestring
+            return attr.decode("utf-8")
```

Comment:
I'm not sure I see how this case happens.

```python
import h5py

f = h5py.File("h5py.h5", "w")
f.attrs["python_bytes"] = b"some bytes"
attr_id = f.attrs.get_id("python_bytes")
h5py.check_string_dtype(attr_id.dtype)
```

```
string_info(encoding='ascii', length=None)
```

This would just hit the previous case, right?

Reply:
```
import h5py
f = h5py.File("test.h5", "w")
f.attrs.create("test", "some bytes", dtype=h5py.h5t.string_dtype(length=15))
h5py.check_string_dtype(f.attrs.get_id("test").dtype)
```
gives
```
 string_info(encoding='utf-8', length=15)
```
and
```
f.attrs["test"]
```
gives 
```
b'some bytes'
```

Reply:
Ahh, yeah, this makes sense, not sure why I didn't get this at the time.